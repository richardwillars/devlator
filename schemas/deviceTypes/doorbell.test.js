const doorbellModule = require("./doorbell");
const constants = require("../../constants");
const eventsModule = require("../events");

const events = eventsModule(constants);

describe("models/doorbell", () => {
  it("should export the commands", () => {
    const doorbellModel = doorbellModule(events, constants);
    expect(doorbellModel.commands).toMatchSnapshot();
  });

  it("should export the events", () => {
    const doorbellModel = doorbellModule(events, constants);
    expect(doorbellModel.events).toMatchSnapshot();
  });
});
