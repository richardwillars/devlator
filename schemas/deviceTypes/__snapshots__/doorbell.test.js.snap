// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models/doorbell should export the commands 1`] = `Object {}`;

exports[`models/doorbell should export the events 1`] = `
Object {
  "properties": Object {
    "BUTTON": Object {
      "constant": "BUTTON",
      "description": "Represents if button is pressed or not",
      "friendly": "Button",
      "responseSchema": Object {
        "properties": Object {
          "pressed": Object {
            "type": "boolean",
          },
        },
        "required": Array [
          "pressed",
        ],
        "type": "object",
      },
      "type": "boolean",
    },
    "MOTION": Object {
      "constant": "MOTION",
      "description": "Represents if motion is detected or not",
      "friendly": "Motion",
      "responseSchema": Object {
        "properties": Object {
          "detected": Object {
            "type": "boolean",
          },
        },
        "required": Array [
          "detected",
        ],
        "type": "object",
      },
      "type": "boolean",
    },
  },
  "type": "object",
}
`;
